### PR TITLE
Update dev Dockerfile to not to require root to build

### PR DESCRIPTION
**Pull request by Victoremepunto**
# Overview

This PR is being created to address [ESSNTL-xxxx](https://issues.redhat.com/browse/ESSNTL-xxxx).
(A description of your PR's changes, along with why/context to the PR, goes here.)

This makes it possible to build the dev.dockerfile Dockerfile with rootless podman

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: dev.dockerfile [hunk 1]
```diff
@@ -1,20 +1,17 @@
 FROM registry.access.redhat.com/ubi8/python-38
 
-USER root
-
+USER 0
```
Comment (thearifismail) [inline dev.dockerfile:3]: Why change to user 0 from 'root'?

File: dev.dockerfile [hunk 2]
```diff
@@ -1,20 +1,17 @@
...
+    dnf clean all
 
-WORKDIR /opt/app-root/src
 COPY . .
+RUN chown -R 1001:0 ./
+USER 1001
 
-RUN pip install --upgrade pip && \
-    pip install pipenv && \
+RUN pip install pipenv==2022.4.8  && \
```
Comment (thearifismail) [inline dev.dockerfile:14]: Why the specific version `2022.4.8` instead of the latest?

**Review by thearifismail — COMMENTED** [review]
@Victoremepunto, though the changes are in dev.dockerfile but trying to understand why are these changes required?

**Comment by Victoremepunto** [comment]
> @Victoremepunto, though the changes are in dev.dockerfile but trying to understand why are these changes required?

I tried to build Inventory's image locally for another task I'm working on, and found that README points to `dev.dockerfile` as the Dockerfile to use for that purpose. Unfortunately I failed to build the image with rootless `podman` so I tried to analyze the root cause and provide a fix, while also refactoring it with what I understand as good practices.

**Comment by thearifismail** [comment]
@Victoremepunto what about using a specific pipenv version, `2022.4.8` instead of the latest?

File: dev.dockerfile (hunk 1 above)
Comment (Victoremepunto) [reply dev.dockerfile:3]: It is just a good practice to be explicit with UIDs or GIDs , see [note](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user)  on the `USER` directive on the official Docker docs

File: README.md [hunk 3]
```diff
@@ -309,6 +309,8 @@ s2i build . centos/python-36-centos7 inventory -e ENABLE_PIPENV=true
 In addition, a [Dockerfile](./dev.dockerfile) is provided.
 The container image built this way is only intended for development purposes (e.g. orchestrating the container using docker-compose) and must not be used for production deployment.
 
+**Note** some of the packages require a subscription, make sure the host building the image is attached to a valid subscription providing RHEL
```
[the remark this answers is not in the document]
Comment (Victoremepunto) [reply README.md:312]: For me it was required for both `postgresql` and `snappy` . I think @RonnyPfannschmidt  hit this issue too

**Comment by Victoremepunto** [comment]
> @Victoremepunto what about using a specific pipenv version, `2022.4.8` instead of the latest?

see https://github.com/RedHatInsights/insights-host-inventory/pull/1123#discussion_r864727403

File: dev.dockerfile (hunk 2 above)
Comment (thearifismail) [reply dev.dockerfile:14]: Building image manually or explicitly using dev.dockerfile is not something we do frequently and haven't felt the need to play with.

**Comment by thearifismail** [comment]
@Victoremepunto since `dev.dockerfile` is used for dev works, can you please add these reasons as comments to this file to help the future users of this file.  Thanks.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kruai) [reply README.md:312]: I can confirm that postgresql does, at the very least

**Comment by Victoremepunto** [comment]
> @Victoremepunto since `dev.dockerfile` is used for dev works, can you please add these reasons as comments to this file to help the future users of this file. Thanks.

Added a comment about why pinning the pip version [here](https://github.com/RedHatInsights/insights-host-inventory/pull/1123/files#diff-ed68d1e304cee11b93a4fd8622e177d16fb23b12f8a835929cf2948c9074eb35R14-R15)

**Review by thearifismail — APPROVED** [review]

**Review by kruai — APPROVED** [review]

**Comment by kruai** [comment]
The IQE test failure appears to be unrelated to these changes. We'll retry a few times and merge it in when it passes.

**Pull request merged**